### ISSUE TITLE
StandaloneAppActivation test failing in CI

### ISSUE THREAD

**Issue by dagood** [dotnet/core-setup#6361]
It looks like the test added in https://github.com/dotnet/core-setup/pull/6325 is blocking PR validation:

https://github.com/dotnet/core-setup/pull/6356 is hitting this:
https://dev.azure.com/dnceng/public/_build/results?buildId=181959&view=logs&jobId=cb452c53-f418-5db4-86fa-6100fffaa4b0&taskId=dd6500e1-1be8-556c-8a79-cfb514756ed1&lineStart=1182&lineEnd=1183&colStart=1&colEnd=1

```
  Failed   Microsoft.DotNet.CoreSetup.Test.HostActivation.StandaloneAppActivation.Running_AppHost_with_GUI_Doesnt_Report_Errors_In_Window_When_Disabled
  Error Message:
   The command error output did not contain expected result: "This executable is not bound to a managed DLL to execute.""
File Name: D:\a\1\s\bin\tests\win-x64.Release\9\StandaloneApp\bin\StandaloneApp.exe
Arguments: 
Exit Code: -2147450731
StdOut:

StdErr:

"
  Stack Trace:
     at FluentAssertions.Execution.XUnit2TestFramework.Throw(String message) in C:\projects\fluentassertions-vf06b\Src\FluentAssertions.Net45\Execution\XUnit2TestFramework.cs:line 32
     at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args) in C:\projects\fluentassertions-vf06b\Src\Core\Execution\AssertionScope.cs:line 197
     at Microsoft.DotNet.CoreSetup.Test.CommandResultAssertions.HaveStdErrContaining(String pattern) in D:\a\1\s\src\test\TestUtils\Assertions\CommandResultAssertions.cs:line 89
     at Microsoft.DotNet.CoreSetup.Test.HostActivation.StandaloneAppActivation.Running_AppHost_with_GUI_Doesnt_Report_Errors_In_Window_When_Disabled() in D:\a\1\s\src\test\HostActivationTests\StandaloneAppActivation.cs:line 359
```

Another PR: https://dev.azure.com/dnceng/public/_build/results?buildId=181885&view=logs

```
  Failed   Microsoft.DotNet.CoreSetup.Test.HostActivation.StandaloneAppActivation.Running_AppHost_with_GUI_Doesnt_Report_Errors_In_Window_When_Disabled
  Error Message:
   The command error output did not contain expected result: "This executable is not bound to a managed DLL to execute.""
File Name: D:\a\1\s\bin\tests\win-x86.Debug\9\StandaloneApp\bin\StandaloneApp.exe
Arguments: 
Exit Code: -2147450731
StdOut:

StdErr:

"
  Stack Trace:
     at FluentAssertions.Execution.XUnit2TestFramework.Throw(String message) in C:\projects\fluentassertions-vf06b\Src\FluentAssertions.Net45\Execution\XUnit2TestFramework.cs:line 32
     at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args) in C:\projects\fluentassertions-vf06b\Src\Core\Execution\AssertionScope.cs:line 197
     at Microsoft.DotNet.CoreSetup.Test.CommandResultAssertions.HaveStdErrContaining(String pattern) in D:\a\1\s\src\test\TestUtils\Assertions\CommandResultAssertions.cs:line 87
     at Microsoft.DotNet.CoreSetup.Test.HostActivation.StandaloneAppActivation.Running_AppHost_with_GUI_Doesnt_Report_Errors_In_Window_When_Disabled() in D:\a\1\s\src\test\HostActivationTests\StandaloneAppActivation.cs:line 358
```

@vitek-karas

**Comment by vitek-karas** [comment]
#6363 should fix the CI.
I'll keep this issue to track the issue with the original test validation sometimes failing.

**Comment by vitek-karas** [comment]
This should be hopefully fixed by #6823 ... please reopen if this still happens.
